### PR TITLE
Fix formatting of chained boolean operations

### PR DESCRIPTION
**Pull request by zanieb**
Closes https://github.com/astral-sh/ruff/issues/6068

These commits are kind of a mess as I did some stumbling around here. 

Unrolls formatting of chained boolean operations to prevent nested grouping which gives us Black-compatible formatting where each boolean operation is on a new line.

### CODE REVIEW

File: crates/ruff_python_formatter/tests/snapshots/black_compatibility@simple_cases__expression.py.snap [hunk 1]
```diff
@@ -351,13 +369,13 @@ True or False
...
 Name1 and Name2 or Name3
-Name1 or (Name2 and Name3)
+Name1 and Name2 or Name3
+Name1 or Name2 and Name3
 Name1 or Name2 and Name3
-(Name1 and Name2) or (Name3 and Name4)
 Name1 and Name2 or Name3 and Name4
-Name1 or (Name2 and Name3) or Name4
+Name1 and Name2 or Name3 and Name4
+Name1 or Name2 and Name3 or Name4
```
Comment (zanieb) [inline crates/ruff_python_formatter/tests/snapshots/black_compatibility@simple_cases__expression.py.snap:378]: Need to fix these

File: crates/ruff_python_formatter/src/expression/expr_bool_op.rs [hunk 2]
```diff
@@ -51,20 +77,35 @@ impl FormatNodeRule<ExprBoolOp> for FormatExprBoolOp {
...
-                    f,
-                    [
-                        op.format(),
-                        space(),
-                        in_parentheses_only_group(&value.format())
-                    ]
-                )?;
+                write!(f, [op.format(), space(),])?;
+
+                match value {
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/src/expression/expr_bool_op.rs:82]: The two matches seem identical. Can we move the logic to a FormatValue { expr } struct that implements Format?

**Review by MichaReiser — APPROVED** [review]

**Comment by MichaReiser** [comment]
🙏

File: crates/ruff_python_formatter/src/expression/expr_bool_op.rs (hunk 2 above)
Comment (zanieb) [reply crates/ruff_python_formatter/src/expression/expr_bool_op.rs:82]: Sure!

**Pull request merged**